prowler/providers/azure/azure_provider.py

```diff
@@ -451,35 +493,74 @@ def test_connection(
+                raise no_auth_method_error
+            return Connection(error=no_auth_method_error)
+        except AzureBrowserAuthNoTenantIDError as browser_no_tenant_error:
+            logger.error(str(browser_no_tenant_error))
+            if raise_on_exception:
+                raise browser_no_tenant_error
+            return Connection(error=browser_no_tenant_error)
+        except AzureTenantIDNoBrowserAuthError as tenant_no_browser_error:
+            logger.error(str(tenant_no_browser_error))
+            if raise_on_exception:
+                raise tenant_no_browser_error
+            return Connection(error=tenant_no_browser_error)
+        # Exceptions from setup_region_config
+        except AzureArgumentTypeValidationError as type_validation_error:
+            logger.error(str(type_validation_error))
+            if raise_on_exception:
+                raise type_validation_error
+            return Connection(error=type_validation_error)
+        except AzureSetUpRegionConfigError as region_config_error:
+            logger.error(str(region_config_error))
+            if raise_on_exception:
+                raise region_config_error
+            return Connection(error=region_config_error)
+        # Exceptions from setup_session
+        except AzureEnvironmentVariableError as environment_credentials_error:
+            logger.error(str(environment_credentials_error))
+            if raise_on_exception:
+                raise environment_credentials_error
+            return Connection(error=environment_credentials_error)
+        except AzureDefaultAzureCredentialError as default_credentials_error:
+            logger.error(str(default_credentials_error))
+            if raise_on_exception:
+                raise default_credentials_error
+            return Connection(error=default_credentials_error)
+        except AzureInteractiveBrowserCredentialError as interactive_browser_error:
+            logger.error(str(interactive_browser_error))
+            if raise_on_exception:
+                raise interactive_browser_error
+            return Connection(error=interactive_browser_error)
+        # Exceptions from SubscriptionClient
+        except HttpResponseError as http_response_error:
             logger.error(
-                f"{validation_error.__class__.__name__}[{validation_error.__traceback__.tb_lineno}]: {validation_error}"
+                f"{http_response_error.__class__.__name__}[{http_response_error.__traceback__.tb_lineno}]: {http_response_error}"
             )
             if raise_on_exception:
-                raise validation_error
-            return Connection(error=validation_error)
-
+                raise AzureHTTPResponseError(
+                    file=os.path.basename(__file__),
+                    original_exception=http_response_error,
+                )
+            return Connection(error=http_response_error)
         except SystemExit as exit_error:
```

Comment:
We should replace this with our exceptions, we should not raise `SystemExit` anymore.

Reply:
True, it shouldn't raises SystemExit never.